### PR TITLE
Adding offbeam gate counting in CAFs

### PR DESCRIPTION
**Pull request by jzettle**
Added offbeam gate counting in CAFs adding a similar object to what is added for BNB and NuMI spill accounting. Looks at the number of gates for BNB and NuMI offbeam alone for offbeam data which event-by-event is computed as the difference in the number of gates for the previous event.

### CODE REVIEW

**Comment by jzettle** [comment]
This is a set of PRs that spans https://github.com/SBNSoftware/sbnana/pull/89, https://github.com/SBNSoftware/sbncode/pull/298, and https://github.com/SBNSoftware/sbnobj/pull/67

File: sbnanaobj/StandardRecord/SREXTInfo.cxx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+#include <climits>
+
+namespace caf
+{
+  SREXTInfo::SREXTInfo():
+    gates_since_last_trigger(std::numeric_limits<unsigned int>::max()),
+    isBNBOffBeam(false),
+    isNuMIOffBeam(false),
+    isMajority(false),
+    isMinBias(false)
```
Comment (jzennamo) [inline sbnanaobj/StandardRecord/SREXTInfo.cxx:13]: These should be commented what they mean or their name should be changed to something more generic.

File: sbnanaobj/StandardRecord/SREXTInfo.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+{
+  class SREXTInfo
+  {
+  public:
+    SREXTInfo();
+    unsigned int gates_since_last_trigger;
+    bool isBNBOffBeam;
+    bool isNuMIOffBeam;
+    bool isMajority;
+    bool isMinBias;
```
Comment (jzennamo) [inline sbnanaobj/StandardRecord/SREXTInfo.h:14]: These should be commented what they mean or their name should be changed to something more generic.

**Review by PetrilloAtWork — CHANGES_REQUESTED** [review]
Fortunately the integration test caught the error somehow...
This commit is trying to redefine the checksum of `caf::SRHeader` version 17 which was previously there. This must not be, since version 17 was seen in a release and data may have been produced with it.
A new version 18 should be added instead:
```xml
<class name="caf::SRHeader" ClassVersion="18">
  <version ClassVersion="18" checksum="2118058550"/>
  <version ClassVersion="17" checksum="2390355049"/>
```

**Pull request closed without merging**